corehq/motech/repeaters/migrations/0008_add_repeater_name.py

```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.16 on 2022-11-18 13:27
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('repeaters', '0007_migrate_to_sql'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='sqlrepeater',
+            name='name',
+            field=models.CharField(max_length=64, null=True),
+        ),
+    ]
```

Comment:
Could we include `populate_repeater_names` as a `call_command` action here in the migration so that the data migration will be part of actual migration? See for e.g. https://github.com/dimagi/commcare-hq/search?q=call_command

Reply:
We _could_. The reason why I did not do it (and want to call this command in the next PR) is because it has the potential to take a long time. I'm open for suggestions

Reply:
I think adding it in migrations would be fine because we have only around 1.5K repeater objects on prod and on other environments number is much lower. So the migration would be fairly quick.

Reply:
Cool, then I'll update the PR. Thanks!

Reply:
On prod there is currently about 2300 repeaters. I created 4000 repeaters locally and the migration ran about 2 minutes 37 seconds. If this is acceptable, then we can keep the data migration in as a migration

Reply:
Sounds okay to me Chris. You may be able to find alternate solutions in https://commcare-hq.readthedocs.io/migrations.html as well.

Reply:
> My thought is to rather manually run the data migration after this PR is deployed. It may still happen that a Repeater is created during these two deployments, but if I run the data migration in the next PR again (but this time as a migration), then it will automatically catch those last few Repeaters without names before the code is going live.

Would we be suggesting the same process for third-party hosters? If yes, then we would have to go through the entire process of releasing a changelog.

For commcare Dimagi environments, you could speed up deployments by 1-2 minutes if you run the command before and then run it again in migrations. But for third-party hosters, would there be any issue if we skip manually running the migrations and just run them as migrations in the next PR which adds the functionality to edit repeater names?

Reply:
Good point. Third party environments usually are single project based, so the number of repeaters should be well within hundred, so it should be okay for it to be part of just the migration/deploy.

Reply:
There would be no risk of running the data migration the first time as part of the branch that adds the editable repeater code, except that it might take longer for third parties to run. As @sravfeyn pointed out, third parties generally have hundreds of repeaters, so this should not be an issue.